src/main/java/com/google/api/codegen/viewmodel/DynamicLangXApiView.java

```diff
@@ -128,6 +128,8 @@ public boolean hasMissingDefaultOptions() {
   @Nullable
   public abstract String grpcTransportImportName();
 
+  public abstract boolean isRestOnlyTransport();
```

Comment:
As per go/actools-regapic-design, in the final product, generated GAPICs must be able to support multiple transports if supported by the API. For Java, we'll support this in the microgenerator; the monolith generates single-transport GAPICs. For PHP, the situation is likely similar, though on a longer timescale. All this to say that I would appreciate structuring the code in terms of "supports REST" rather than "REST only", but only where this doesn't require undue work. For everything else, please add a TODO to the effect of  "we're generating single-transport GAPICs for now, though in the fullness of time we'll generate multi-transport GAPICs."

Reply:
It doesn't make sense to say `supports REST` because PHP GAPICs already support REST, and we do not need any special handling (REGAPIC). 

I could change this to `isDireGapic()`, `isDiscovery()`, or something along these lines, but it's very important to distinguish from existing REST support in OnePlatform GAPICs and the support for DIscovery REST-only APIs.

Reply:
Looks like the gist of the problem is we are either missing `GRPC_HTTP` value in the `TransportProtocol` enum or we should convert all usages of `TransportProtocol` to be a list value, which may contain multiple entries (to represent grpc+rest).

In that case you would not need `isRestOnlyTransport` the `TransportProtocol` value will contain the information you need. This is basically technical debt introduced for java started biting PHP implementation.

Reply:
Yes, the eventual goal is to allow list values, so that `grpc+rest` works. I would love it if support for the flags could be made to work now, and I would love it even more if the implementation supported multi-transport.

One question, then, about PHP: since PHP already supports REST, REGAPIC is not replacing the existing support, but just adding to it, right? Am I missing something?

Reply:
I still think, rather than focus on the `TransportProtocol` as the means to derive this logic, we should remove `TransportProtocol` as an enum/option/parameter and instead use `ClientType`. This would allow us to have `ClientType.GAPIC`, `ClientType.REGAPIC`, and `ClientType.DIREGAPIC` (I would prefer `ClientType.DISCOVERY` and `ClientType.ONE_PLATFORM`, but if you feel differing between GAPIC and REGAPIC is important we can do the former).

Reply:
We could also do, instead, something along the lines of `hasGrpcSupport`, which serves the same purpose as `isRestOnly`, but makes it a little clearer. WDYT?

Reply:
Changed to `supportsGrpcTransport`